LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDeviceClient.cs

```diff
@@ -156,61 +177,31 @@ public async Task<Message> ReceiveAsync(TimeSpan timeout)
-
-                return true;
-            }
-            catch (OperationCanceledException ex) when (ExceptionFilterUtility.True(() => this.logger.LogError($"could not complete cloud to device message (id: {cloudToDeviceMessage.MessageId ?? "undefined"}) with error: {ex.Message}")))
-            {
-                return false;
-            }
-        }
-
-        public async Task<bool> AbandonAsync(Message cloudToDeviceMessage)
-        {
-            if (cloudToDeviceMessage is null) throw new ArgumentNullException(nameof(cloudToDeviceMessage));
-
-            try
-            {
-                this.logger.LogDebug($"abandoning cloud to device message, id: {cloudToDeviceMessage.MessageId ?? "undefined"}");
+        public Task<bool> CompleteAsync(Message cloudToDeviceMessage) =>
+            ExecuteC2DOperationAsync(cloudToDeviceMessage, static (client, message) => client.CompleteAsync(message), CompleteOperationName);
 
-                await this.deviceClient.AbandonAsync(cloudToDeviceMessage);
+        public Task<bool> AbandonAsync(Message cloudToDeviceMessage) =>
+            ExecuteC2DOperationAsync(cloudToDeviceMessage, static (client, message) => client.AbandonAsync(message), AbandonOperationName);
 
-                this.logger.LogDebug($"done abandoning cloud to device message, id: {cloudToDeviceMessage.MessageId ?? "undefined"}");
+        public Task<bool> RejectAsync(Message cloudToDeviceMessage) =>
+            ExecuteC2DOperationAsync(cloudToDeviceMessage, static (client, message) => client.RejectAsync(message), RejectOperationName);
 
-                return true;
-            }
-            catch (OperationCanceledException ex) when (ExceptionFilterUtility.True(() => this.logger.LogError($"could not abandon cloud to device message (id: {cloudToDeviceMessage.MessageId ?? "undefined"}) with error: {ex.Message}")))
-            {
-                return false;
-            }
-        }
-
-        public async Task<bool> RejectAsync(Message cloudToDeviceMessage)
+        private async Task<bool> ExecuteC2DOperationAsync(Message cloudToDeviceMessage, Func<DeviceClient, Message, Task> executeAsync, string operationName)
         {
             if (cloudToDeviceMessage is null) throw new ArgumentNullException(nameof(cloudToDeviceMessage));
+            var messageId = cloudToDeviceMessage.MessageId ?? "undefined";
 
             try
             {
-                this.logger.LogDebug($"rejecting cloud to device message, id: {cloudToDeviceMessage.MessageId ?? "undefined"}");
-
-                await this.deviceClient.RejectAsync(cloudToDeviceMessage);
+                this.logger.LogDebug("'{OperationName}' cloud to device message, id: '{MessageId}'.", operationName, messageId);
+                using var rejectOperation = this.tracing.TrackIotHubDependency(GetSdkDependencyName(operationName), $"{this.deviceIdTracingData}&messageId={messageId}");
```

Comment:
Shouldn't this variable be called something more generic instead of `rejectOperation`? Maybe simply `operation`?

Reply:
Renamed in 8be5d46ef764f1d18655bf159c53ee6cd7cefd25